webui/src/lib/components/repository/changes.jsx

```diff
@@ -199,6 +252,31 @@ export const TreeEntryPaginator = ({ path, setAfterUpdated, nextPage, depth=0, l
     );
 };
 
+export const DeltaTableChangesTreeItem = ({ entry, repo, reference, leftDiffRefID, rightDiffRefID, internalRefresh, onRevert, onNavigate, delimiter, relativeTo, getMore}) => {
+    return <>
+        <TreeItem key={entry.path + "-item"} entry={entry} repo={repo}
+                  changesTreeType={ChangesTreeType.DeltaLakeTableChanges}
+                  reference={reference}
+                  internalReferesh={internalRefresh} leftDiffRefID={leftDiffRefID}
+                  rightDiffRefID={rightDiffRefID} delimiter={delimiter}
+                  relativeTo={relativeTo}
+                  onNavigate={onNavigate}
+                  getMore={getMore}/>
+    </>
+}
+
+export const ObjectChangesTreeItem = ({ entry, repo, reference, leftDiffRefID, rightDiffRefID, internalRefresh, onRevert, onNavigate, delimiter, relativeTo, getMore}) => {
+    return <>
+        <TreeItem key={entry.path + "-item"} entry={entry} repo={repo}
+                  reference={reference}
+                  internalReferesh={internalRefresh} leftDiffRefID={leftDiffRefID}
+                  rightDiffRefID={rightDiffRefID} delimiter={delimiter}
+                  relativeTo={relativeTo}
+                  onNavigate={onNavigate}
+                  getMore={getMore}/>
+    </>
+}
```

Comment:
it's because 
> Hooks cannot be called conditionally (unless you wrap them in a component)

Tree item calls some hooks conditionally. not a good reason but this is the explanation here 

from: https://gofore.com/en/advanced-react-hooks-the-strategy-pattern/